kubetest/gke.go

```diff
@@ -566,8 +591,15 @@ func (g *gkeDeployer) Down() error {
 		"gcloud", g.containerArgs("clusters", "delete", "-q", g.cluster,
 			"--project="+g.project,
 			g.location)...))
-	errFirewall := control.FinishRunning(exec.Command("gcloud", "compute", "firewall-rules", "delete", "-q", firewall,
-		"--project="+g.project))
+	var errFirewall error
+	if control.FinishRunning(exec.Command("gcloud", "compute", "firewall-rules", "describe", firewall,
+		"--project="+g.project,
+		"--format=value(name)")) == nil {
+		//delete firewall only if exists
+		errFirewall = control.FinishRunning(exec.Command("gcloud", "compute", "firewall-rules", "delete", "-q", firewall,
```

Comment:
Sorry I might be missing something, I see `cleanupNetworkFirewalls()` is deleting firewall from a specific network. What is this for (deleting firewalls from default network)?

Reply:
I assume that this specific network is something that was created for only testing purpose, so it should be removed after all test are finished (this logic is already in the code). At this point some tests leak resources (firewall rules) which results in those test networks not being removed.
With cleanupNetworkFirewalls() I want to ensure that all resources are being cleaned up.
Removing resources is important because if there are some resources left, it will affect other test executions.

Reply:
Thanks that makes sense. Now I see the firewall comes from `getClusterFirewall()`. Keeping this separate seems good.

Minor: might be better to have a more specific comment other than "delete firewall only if exists" :)